core/src/main/scala/akka/persistence/r2dbc/internal/postgres/PostgresDurableStateDao.scala

```diff
@@ -64,15 +67,17 @@ private[r2dbc] object PostgresDurableStateDao {
       binding: AdditionalColumn.Binding[_])
 
   val FutureDone: Future[Done] = Future.successful(Done)
+  val FutureInstantNone: Future[Option[Instant]] = Future.successful(None)
 }
 
 /**
  * INTERNAL API
  */
 @InternalApi
-private[r2dbc] class PostgresDurableStateDao(settings: R2dbcSettings, connectionFactory: ConnectionFactory)(implicit
-    ec: ExecutionContext,
-    system: ActorSystem[_])
+private[r2dbc] class PostgresDurableStateDao(
+    settings: R2dbcSettings,
+    connectionFactory: ConnectionFactory,
+    journalDao: JournalDao)(implicit ec: ExecutionContext, system: ActorSystem[_])
```

Comment:
Would it be worth making the journal-dao creation/dependency more lazy, for use cases where only using "regular" durable state without the change events, this probably means picking up/requiring journal config and maybe even more runtime stuff?

Reply:
Two options:
1. reuse the same instance of the JournalDao inside the PostgresDialect
2. pass in the dialect as parmeter here so that it can create the JournalDao on demand

Reply:
Right now we create a separate instance when constructing the durable state dao, can we instead pass a factory lambda and have the journalDao as a lazy field initialized on first use (if that ever happens)?

Reply:
changed in https://github.com/akka/akka-persistence-r2dbc/pull/485/commits/b0b535da3b77461021ce3fb1fc08fa1b7bd6139b